Kerberos support for presto

anishanainani opened the pull request:
Added Kerberos support in presto.py script

- Added auth parameter to the Cursor
- By default auth is NONE but user can pass NONE/PLAIN and KERBEROS

prabhu1984 commented:
Great to see Kerberos support for presto! Can you please review and merge?

jingw commented:
Sorry, haven't forgotten about this. Will take a look next week.

devinstevenson commented on pyhive/presto.py:
```diff
@@ -79,7 +80,7 @@ class Cursor(common.DBAPICursor):
     def __init__(self, host, port='8080', username=None, catalog='hive',
                  schema='default', poll_interval=1, source='pyhive', session_props=None,
-                 protocol='http', password=None):
+                 protocol='http', password=None,auth='NONE',verify=False):
```
wouldn't it be best to have as default `verify=True`? deactivating SSL should be explicitly done.

anishanainani replied:
I put it as False because usually users would use it without SSL but I can set it to True by default if that is preferred.

devinstevenson commented:
@anishanainani adding `requests_kerberos` to `tests_require` and `requires_extra` in `setup.py` should fix the test failures.

anishanainani commented:
@devinstevenson I added requests_kerberos to tests_require and requires_extra in setup.py, but the tests are still failing with requests_kerberos module not found. Can you please review this? Thanks

devinstevenson commented:
@anishanainani sorry, you need to add it to dev_requirements.txt file too.

anishanainani commented:
@devinstevenson I added requests_kerberos in dev_requirements and did some small fixes. All the tests failures are now fixed. Can you please review this? Thanks

jingw commented on pyhive/presto.py:
```diff
@@ -18,6 +18,7 @@
 import requests
 from requests.auth import HTTPBasicAuth
 
+from requests_kerberos import HTTPKerberosAuth, OPTIONAL
```
Could you make this an optional dependency?
```
try:
    import requests_kerberos
except ImportError:
    # no kerberos support
    pass
```

jingw replied:
above comment is obsolete assuming we just go with the `requests_options` approach.

jingw commented on pyhive/presto.py:
```diff
@@ -79,7 +80,7 @@ class Cursor(common.DBAPICursor):
     def __init__(self, host, port='8080', username=None, catalog='hive',
                  schema='default', poll_interval=1, source='pyhive', session_props=None,
-                 protocol='http', password=None):
+                 protocol='http', password=None, auth='NONE', verify=True):
```
When would someone want to pass verify=False?

anishanainani commented:
@jingw Thanks. I think thats a good idea. We should close this Pull request then and I can send a new one with requests_options.

jingw commented:
I'm going to push a change to make requests options customizable.
